README.md

```diff
@@ -1,63 +1,142 @@
 # <img src="https://docs.lido.fi/img/logo.svg" alt="Lido" width="46"/> Lido Oracle
 
-[![Tests](https://github.com/lidofinance/lido-oracle/workflows/Tests/badge.svg?branch=daemon_v2)](https://github.com/lidofinance/lido-oracle/actions)
 [![License: GPL v3](https://img.shields.io/badge/License-GPLv3-blue.svg)](https://www.gnu.org/licenses/gpl-3.0)
+[![Tests](https://github.com/lidofinance/lido-oracle/workflows/Tests/badge.svg?branch=daemon_v2)](https://github.com/lidofinance/lido-oracle/actions)
+
+Oracle daemon for Lido decentralized staking service: Monitoring the state of the protocol across both layers and submitting regular update reports to the Lido smart contracts.
+
+## How it works
+
+There are three modules in the oracle:
+
+- Accounting
+- Ejector
+
+### Accounting module
+
+Accounting module updates the protocol TVL, distributes node-operator rewards, and processes user withdrawal requests.
+
+**Flow**
+
+The oracle work is delineated by time periods called frames. Oracles finalize a report in each frame.
+The default Accounting Oracle frame length is 225 epochs, which is 24 hours. The frame includes these stages:
+
+- **Waiting** - oracle starts as daemon and wakes up every 12 seconds (by default) in order to find the last finalized slot (ref slot).
+  If ref slot missed, Oracle tries to find previous non-missed slot.
+- **Data collection**: oracles monitor the state of both the execution and consensus layers and collect the data;
+- **Hash consensus**: oracles analyze the data, compile the report and submit its hash to the HashConsensus smart contract;
+- **Core update report**: once the quorum of hashes is reached, meaning more than half of the oracles submitted the same hash,
+  one of the oracles chosen in turn submits the actual report to the AccountingOracle contract, which triggers the core protocol
+  state update, including the token rebase, distribution of node operator rewards, finalization of withdrawal requests, and
+  deciding whether to go in the bunker mode.
+- **Extra data report**: an additional report carrying additional information. This part is required.
+  All rewards to modules are distributed in this Oracle’s phase. Can be submitted in chunks.
```

Comment:
> All rewards to modules

Rewards are allocated to modules in the second stage, and to node operators in the third stage

> Can be submitted in chunks

This is planned in the future, but so far only one transaction is allowed